### PR TITLE
feat: Add support for all resources of k8s and for production server

### LINKED ISSUE #13: The automated release is failing 🚨
## :rotating_light: The automated release from the `master` branch failed. :rotating_light:

I recommend you give this issue a high priority, so other packages depending on you can benefit from your bug fixes and new features again.

You can find below the list of errors reported by **semantic-release**. Each one of them has to be resolved in order to automatically publish your package. I’m sure you can fix this 💪.

Errors are usually caused by a misconfiguration or an authentication problem. With each error reported below you will find explanation and guidance to help you to resolve it.

Once all the errors are resolved, **semantic-release** will release your package the next time you push a commit to the `master` branch. You can also manually restart the failed CI job that runs **semantic-release**.

If you are not sure how to resolve this, here are some links that can help you:
- [Usage documentation](https://github.com/semantic-release/semantic-release/blob/caribou/docs/usage/README.md)
- [Frequently Asked Questions](https://github.com/semantic-release/semantic-release/blob/caribou/docs/support/FAQ.md)
- [Support channels](https://github.com/semantic-release/semantic-release#get-help)

If those don’t help, or if this issue is reporting something you think isn’t right, you can always ask the humans behind **[semantic-release](https://github.com/semantic-release/semantic-release/issues/new)**.

---

### Missing `package.json` file.

A [package.json file](https://docs.npmjs.com/files/package.json) at the root of your project is required to release on npm.

Please follow the [npm guideline](https://docs.npmjs.com/getting-started/creating-node-modules) to create a valid `package.json` file.

---

Good luck with your project ✨

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

### PR DESCRIPTION
**Pull request by ashish493**
Fix: https://github.com/casbin/k8s-authz/issues/13

Signed-off-by: Ashish Malik <b218008@iiit-bh.ac.in>

### CODE REVIEW

**Comment by ashish493** [comment]
@hsluoyz , This branch doesn't have the changes from the previous PR. Should I merge the changes with my master branch or should I wait for you to merge the PR #11 and then pull the changes from the casbin's master branch?

File: README.md [hunk 1]
```diff
@@ -44,6 +44,14 @@ Before proceeding, make sure to have the following-
 ```
 kubectl apply -f deployment.yaml
 ```
+- For a production server, we need to create a k8s `secret` to place the certificates for security purposes. 
+```
+kubectl create secret generic casbin -n default \
+  --from-file=key.pem=certs/casbin-key.pem \
+  --from-file=cert.pem=certs/casbin-crt.pem
+```
```
Comment (ashish493) [inline README.md:52]: This was done in order to make the middleware to work on production grade servers.

File: manifests/deployment.yaml [hunk 2]
```diff
@@ -60,8 +60,8 @@ webhooks:
         path: "/validate"
       caBundle: "${CA_BUNDLE}"
     rules:
-      - operations: ["CREATE","UPDATE","DELETE","CONNECT"]
+      - operations: ["*"]
         apiGroups: [""]
         apiVersions: ["v1"]
-        resources: ["pods"]
+        resources: ["*/*"]
```
Comment (ashish493) [inline manifests/deployment.yaml:66]: These changes indicate that, when a user performs any type of operation on any type of resources or subresources of k8s, this webhook will get triggered.

**Comment by hsluoyz** [comment]
@ashish493 sorry for delay, I have merged: https://github.com/casbin/k8s-authz/pull/11 , plz rebase and update this PR.

**Comment by ashish493** [comment]
@hsluoyz Could u review the changes?

File: .releaserc.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "debug": true,
+    "branches": [
```
Comment (ashish493) [inline .releaserc.json:3]: @hsluoyz , I hope this would fix #13.

**Comment by hsluoyz** [comment]
@ashish493 this PR's files are not correct: https://github.com/casbin/k8s-authz/pull/12/files , most of the diffs are already in master. Can you rebase it to master HEAD?

image

image

**Comment by ashish493** [comment]
Moved to #14

**Pull request closed without merging**